Develop

apologet-insane opened the pull request:
Тесты готовы. 
Папка allure-results запушена.

apologet-insane committed:
add allure report

apologet-insane committed:
Тесты готовы

apologet-insane committed:
Тесты готовы. Allure готов

egorvas commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
```
baseURI лучше выносить в одно общее место для всех тестов. На практике твои тесты будут запускаться на разных энвах: дев, прод, тест и тп и везде будут разные url. По этому чтобы в каждом тесте не менять baseURI лучше это делать один раз для всех тестов.

egorvas commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+    @DisplayName("Проверка создания курьера: код ответа и тело: ok:true")
+    public void createCourierPosistive(){
+
+        String registerRequestBody = "{\"login\":\"" + courierLogin + "\","
```
Я считаю что такое формирования body запроса это очень плохая практика(да, я знаю что в задании указано так, мы работаем на этим :-) ). 
Лучше формировать JSON через сериализацию объеков с помощью специальных модулей(jackson или gson), делать из стандартного Java класса HashMap  или же с помощью спецтальных модулей вроде json и его класса JSONObject.
Формирования body из строк с кучей обратных слешей, плюсиков и кавычек выглядит громостко и не по программистски :)

egorvas commented on src/test/java/CourierLoginTest.java:
```diff
@@ -0,0 +1,209 @@
+                .post("/api/v1/courier/login")
+                .then().assertThat().statusCode(200).and().extract().path("id");
+        //удаляем тестового курьера
+        String deleteCourier = "/api/v1/courier/"+id;
```
Все что не связано с логикой теста должно быть вынесено в @After и @Before функции

egorvas commented on src/test/java/OrdersListTest.java:
```diff
@@ -0,0 +1,69 @@
+                .and().extract().jsonPath().getList("orders");
+
+       //проверяем, что список не пустой
+       assertFalse(orders.isEmpty());
```
Чтобы убедится что у тебя возвращается реальный список заказов, а не просто список пустых элементов вроде [{},{}] проверяй в этом месте хотябы наличие id заказов.

apologet-insane committed:
1. Респонсы и body в степах
2. Определяется баг с логином курьера
3. Степы видно в аллюре
4. Исправлено падение при формировании заказов из-за даты

apologet-insane committed:
1. Респонсы и body в степах
2. Определяется баг с логином курьера
3. Степы видно в аллюре
4. Исправлено падение при формировании заказов из-за даты
5. Тесты вынесены в степы, потому что так в тренажёре (надо вернуть)
6. Аллюр закомичен

apologet-insane committed:
1. Респонсы и body в степах
2. Определяется баг с логином курьера
3. Степы видно в аллюре
4. Исправлено падение при формировании заказов из-за даты
5. Проверки расположены в степах
6. Аллюр закомичен

apologet-insane committed:
1. Респонсы в степах
2. Тесты в тестах
3. В сериализацию пока не смог, очень громоздкая конструкция получается.
4. Тесты на зависание API есть, но таймаут в документации к RestAssure не нашёл нормальный. Коллеги по курсу вообще закомментировали данные тесты, чтобы не тормозили процесс.
